### PR TITLE
User guide update (includes #32)

### PR DESCRIPTION
**Pull request by climbfuji**
This PR includes the user guide update in #32, fixes the merge conflicts, fixes the formatting of the tables (replace tabs with spaces), and corrects some actual errors in the tables. When this gets merged into the documentation branch, the original PR will be marked as merged, too.

### CODE REVIEW

**Comment by climbfuji** [comment]
Esteemed reviewers, once this is merged I will create a PR to merge the entire documentation branch into ufs_public_release.

File: doc/UsersGuide/source/ContributingDevelopment.rst [hunk 1]
```diff
@@ -4,4 +4,218 @@
...
+   -r use Rocoto workflow manager
+   -e use ecFlow workflow manager
+   -h display this help
+
+.. code-block:: console
+
+   % cd ufs-weather-model/tests
+   % ./rt.sh -f
+
+This command can only be used on a NOAA machine using the Intel compiler, where an official baseline is available. For information on testing the CCPP code, or using alternate computational platforms, see the following sections.
```
Comment (JulieSchramm) [inline doc/UsersGuide/source/ContributingDevelopment.rst]: Can't ./rt.sh be run on non-NOAA machines, i.e. Cheyenne? And with the GNU compiler?

File: doc/UsersGuide/source/ContributingDevelopment.rst [hunk 2]
```diff
@@ -4,4 +4,218 @@
...
+
+This command and all others below produce log output in ./tests/log_machine.compiler. These log files contain information on the location of the run directories that can be used as templates for the user. Each rt*.conf contains one or more compile commands preceding a number of tests.
+
+Regression test log files (ufs-weather-model/tests/Compile_$(MACHINE_ID).log and ufs-weather-model/tests/RegressionTests_$(MACHINE_ID).log ) will be updated.
+
+Regression testing is only possible on machines for which baselines exist, that is Tier 1 and 2 platforms.
```
Comment (JulieSchramm) [inline doc/UsersGuide/source/ContributingDevelopment.rst]: Regression testing is possible on a machine without a baseline. Baselines should be available on Tier 1 and possibly on Tier 2 platforms (please clarify).  ./rt.sh should run on Tier 1 and 2 platforms, in which case, a baseline may be available. If not, one can be created as described below.

File: doc/UsersGuide/source/ContributingDevelopment.rst [hunk 3]
```diff
@@ -4,4 +4,218 @@
...
+Only developers using Tier 1 and 2 platforms can run the ufs-weather-model regression tests. Other developers need to work with the code managers to assure completion of the regression tests.
+
+
+To run regression test using rt.sh
+
+rt.sh is a bash shell file to run the RT and has the following options:
+
+.. code-block:: console
+
+   Usage: $0 -c <model> | -f | -s | -l <file> | -m | -r | -e | -h
```
Comment (DusanJovic-NOAA) [inline doc/UsersGuide/source/ContributingDevelopment.rst]: Please replace ``$0`` with ``./rt.sh``

Comment (DusanJovic-NOAA) [reply doc/UsersGuide/source/ContributingDevelopment.rst]: Also, ``-c`` option does not need ``<model>`` anymore. We might need to update usage message in the script itself.

File: doc/UsersGuide/source/ContributingDevelopment.rst [hunk 4]
```diff
@@ -4,4 +4,218 @@
...
+
+This command and all others below produce log output in ./tests/log_machine.compiler. These log files contain information on the location of the run directories that can be used as templates for the user. Each rt*.conf contains one or more compile commands preceding a number of tests.
+
+Regression test log files (ufs-weather-model/tests/Compile_$(MACHINE_ID).log and ufs-weather-model/tests/RegressionTests_$(MACHINE_ID).log ) will be updated.
+
+Regression testing is only possible on machines for which baselines exist, that is Tier 1 and 2 platforms.
+
+When porting the code to a new machine, it is useful to start by establishing a personal baseline. Future runs of the RT can then be compared against the personal baseline to ascertain that the results have not been inadvertently affected by code developments. The rt.sh -c option is used to create a personal baseline.
+
+./rt.sh -l rt.conf -c fv3 # create own reg. test baseline
```
Comment (DusanJovic-NOAA) [inline doc/UsersGuide/source/ContributingDevelopment.rst]: Remove ``fv3`` after ``-c``

File: doc/UsersGuide/source/ContributingDevelopment.rst (hunk 1 above)
Comment (climbfuji) [reply doc/UsersGuide/source/ContributingDevelopment.rst]: Yes, this is wrong. My job was not to _work_ on the documentation but to merge what is there now. Hmpf. I will correct it.

File: doc/UsersGuide/source/ContributingDevelopment.rst (hunk 2 above)
Comment (climbfuji) [reply doc/UsersGuide/source/ContributingDevelopment.rst]: See above.

File: doc/UsersGuide/source/ContributingDevelopment.rst (hunk 4 above)
Comment (climbfuji) [reply doc/UsersGuide/source/ContributingDevelopment.rst]: Yes, will do. My job was not to _work_ on the documentation but to merge what is there now. Hmpf. I will correct it.

File: doc/UsersGuide/source/ContributingDevelopment.rst (hunk 3 above)
Comment (climbfuji) [reply doc/UsersGuide/source/ContributingDevelopment.rst]: See below.

**Comment by climbfuji** [comment]
@JulieSchramm @DusanJovic-NOAA As I am reading the paragraph starting at line 180, this is wrong! You can't just create a new baseline on a different platform using rt.sh. The script contains machine-specific sections and uses several scripts underneath that contain machine-specific sections.

File: doc/UsersGuide/source/ContributingDevelopment.rst (hunk 2 above)
Comment (climbfuji) [reply doc/UsersGuide/source/ContributingDevelopment.rst]: > @JulieSchramm @DusanJovic-NOAA As I am reading the paragraph starting at line 180, this is wrong! You can't just create a new baseline on a different platform using rt.sh. The script contains machine-specific sections and uses several scripts underneath that contain machine-specific sections.

And it contradicts several other sections in the document.

**Comment by DusanJovic-NOAA** [comment]
> @JulieSchramm @DusanJovic-NOAA As I am reading the paragraph starting at line 180, this is wrong! You can't just create a new baseline on a different platform using rt.sh. The script contains machine-specific sections and uses several scripts underneath that contain machine-specific sections.

And creating new baselines (outputs) you need baseline input data, which is only available on supported platforms. One can potentially copy only baseline input to a new platform, modify rt.sh (and probably rt_utils.sh) to support platform specific module loading, scheduler etc., which is basically what we do when we port rt.sh to a new platform.

**Comment by climbfuji** [comment]
I am also going to rewrite the NEMSCompsetRun part in the end.

File: tests/rt.sh [hunk 5]
```diff
@@ -7,7 +7,7 @@ die() { echo "$@" >&2; exit 1; }
 usage() {
   set +x
   echo
-  echo "Usage: $0 -c <model> | -f | -s | -l <file> | -m | -k | -r | -e | -h"
+  echo "Usage: $0 -c | -f | -s | -l <file> | -m | -k | -r | -e | -h"
   echo
   echo "  -c  create new baseline results for <model>"
```
Comment (DusanJovic-NOAA) [inline tests/rt.sh]: Remove `` for <model>`` from "echo" command.

File: doc/UsersGuide/source/ContributingDevelopment.rst [hunk 6]
```diff
@@ -161,7 +161,7 @@ rt.sh is a bash shell file to run the RT and has the following options:
 
 .. code-block:: console
 
-   Usage: $0 -c <model> | -f | -s | -l <file> | -m | -r | -e | -h
+   Usage: ./rt.sh -c | -f | -s | -l <file> | -m | -r | -e | -h
```
Comment (DusanJovic-NOAA) [inline doc/UsersGuide/source/ContributingDevelopment.rst]: Do we want to document ``-k`` option?

File: tests/rt.sh (hunk 5 above)
Comment (climbfuji) [reply tests/rt.sh]: I did that, I think? Looks like at least.

Comment (climbfuji) [reply tests/rt.sh]: Ok, line 12 . Got it.

**Comment by climbfuji** [comment]
@DusanJovic-NOAA when you got time, can we finalize this PR please? We need this to be merged today, then merge the entire documentation branch into the public release branch before renaming it and creating the tag. Thanks ...

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
Thanks! I don't have permissions to merge this, can someone with superpowers do this, please?

**Pull request merged**